### ISSUE TITLE
UI issues in list view after latest update.

### ISSUE THREAD

**Issue by MohitKandu25** [FlutterFlow/flutterflow-issues#279]
## Expected behavior (required)
1. TexField inside a ListView should retain its focus on the tapped widget.

## Current behavior (required)
1. As soon as I tap on the TextField inside a listview, the focus moves to a different TextField outside the ListView.

## To Reproduce (required)
1. Create a new page
2. Add a Form
3. Add some TextFields in the form
4. Add a ListView inside the form
5. Add some TextFields inside the ListView
6. Run the app and go to this page
7. Tap on the TextField inside the ListView
8. The focus switches to the last TextField outside the ListView

## Context (required)
- This issue is preventing user from being able to fill the form as they're not able to type in the fields

## Screenshots / recordings
https://user-images.githubusercontent.com/104911624/209521996-a06c94c9-11a6-4130-a6ef-eae46ec9285c.mp4

## Your environment
* Version of FlutterFlow used: v3.0 December 24, 2022 - Flutter 3.3.4
* Browser name and version: Google Chrome 103.0.5060.134 (Official Build) (64-bit)
* Operating system and version: Ubuntu 20.04.4 LTS

**Comment by agreaves** [comment]
Hi @MohitKandu25, thank you for the bug report. Could you please include the Bug Report Code for your project (right click anywhere on the widget tree to generate)?

**Comment by MohitKandu25** [comment]
Hello @agreaves. Thank you for the response. Here's the bug report: ITwohvHqy4x1ocNA1pDbbcFFoCoROVA/U40elusbTUgYDOv0POw+Ps/BMlZtXbGmTHxYM2D88X4z0+DXvt2WDu44ZUysfNRY3qt2SAHwTmaWMISJB4eSOnxPJ8JYGmGMwpuw3Qp7NstoL3thwFOqDNiRFz7aJrqyCkQ7G/KeXdeK2SrDX1iXc2URm05KZDPz

**Comment by sumit5101992** [comment]
@agreaves Is there any update on this?

**Comment by MohitKandu25** [comment]
Hi @agreaves. Do you have any update on this? We are not able to release because of this issue. Would really appreciate it if you could get back to us as soon as you can.

**Comment by bulgariamitko** [comment]
Do you have any Actions on the TextFields that are not working?

**Comment by MohitKandu25** [comment]
Hi @bulgariamitko. Thank You for your response. Yes I did. I had update local state action on the onChange event. However, I removed all the actions attached to the text fields. Basically, it was just a text field in a component without any actions on it. But the issue still persists.

**Comment by wenkaifan0720** [comment]
Hi @MohitLandu25. I just created a project following your steps and didn't see the issue (https://app.flutterflow.io/project/flutterflow-issue-279-ugqszs). Is it still happening to you?

**Comment by MohitKandu25** [comment]
Hi @wenkaifan0720. The issue is still there. I wanted to mention a couple of things that I missed in the steps. Apologies for that.
The list items are generated from a local state which is a JSON and the text field takes initial value from component parameter.
I have 2 pages in the app where this issue is present and both have text fields inside list views. 
I am sharing my project with you and I would really appreciate it if you could take some time and take a look at the pages yourself. The pages are named "MarkAssessment", "CreateLessonPlan" and "EditLessonPlan".
project link: https://app.flutterflow.io/project/coaching-center-h7cmfc
Adding you so that you can access the project.
Basically the pages have a list view.
The list view contains components which contain a text field.
These text fields have onChange actions which save the text field value into a local state using the index coming from parameter.

**Comment by wenkaifan0720** [comment]
Hi @MohitKandu25. I couldn't login with the credentials you provided. I am getting some failure message either logging in a coaching center or individual coach. 

I updated the test project (make a wrapper component with a form and a textfield and with the actions) but still can't reproduce the error. Could you try to reproduce the issue in this [project](https://app.flutterflow.io/project/flutterflow-issue-279-ugqszs)? I've added you ([mohit.kandu@iesoftek.com](mailto:mohit.kandu@iesoftek.com)) as editor (hopefully this is your email).

**Comment by MohitKandu25** [comment]
Hello @wenkaifan0720. I am not sure why you weren't able to login as I tried and it worked just fine.
But as you asked, I have re-created the issue in the project that you shared with me. It contains only one page in which the issue is present. The issue won't show up in test or debug mode so please download the apk and test it in phone.

**Comment by wenkaifan0720** [comment]
Hi @MohitKandu25 . I think I have identified the issue. The touble is, the fix seems to be interfering with another fix introduced before. So we need more time to test it out.

I can tell you the fix now and you can try it out and see if it solves the issue without introducing other issues. Could you try to remove the 'UniqueKey()' that is assigned to the custom components inside of that dynamic list? I know this involves modifying the generated code so it should really be a temporary fix.

**Comment by MohitKandu25** [comment]
Hi @wenkaifan0720 . Sure. I shall try your solution and update here.

**Comment by MohitKandu25** [comment]
Hello @wenkaifan0720 . I tested the workaround you mentioned and while it did fix the issue, it introduced another one. It occurs when removing an item from the list. So this is the issue:
If I have 2 sessions in list and I delete the second item then I have the first item left in the list.
I have checked the local state which is binded to this session object and it's updating correctly.
But in the UI I see that second item was removed and the first item remains.
This issue didn't occur when the UniqueKey() was present.

I have added a video showing the above problem following these steps:
1. Add the session details in first session item (session name and activity name). In the video I have added "Session 1" and "Activity 1".
2. Add another session by clicking on "Add another session" button
3. Add some details in second session item (session name and activity name). In the video I have added "Session 2" and "Activity 2".
4. Delete the first session item which should leave the second item only.
5. In the UI we see second item getting removed and first item remains.

video showing the issue: 
https://user-images.githubusercontent.com/104911624/215021209-763f544e-0da0-45f6-9357-6bfbd71729a8.mp4

**Comment by bulgariamitko** [comment]
Maybe I would suggest you to create a new issue about this.

**Comment by wenkaifan0720** [comment]
Hi @MohitKandu25. We are aware of this issue now. This can be fixed by replacing the `UniqueKey` with a `ValueKey` with an appropriate value. We shall deploy this fix soon.

**Comment by wenkaifan0720** [comment]
Hi @MohitKandu25 , the fix has been pushed and there's no need for a new issue. We have accounted for the deletion inside a list. The update just haven't arrived at the public version of FF so you need to wait for a few days.
